packages/jsii-pacmak/test/expected.jsii-calc-base/python/src/scope/jsii_calc_base/__init__.py

```diff
@@ -23,18 +23,44 @@ def __init__(self) -> None:
     @jsii.member(jsii_name="typeName")
     def type_name(self) -> typing.Any:
         """
-        Returns:
-            the name of the class (to verify native type names are created for derived classes).
+        return
+        :return: the name of the class (to verify native type names are created for derived classes).
         """
         return jsii.invoke(self, "typeName", [])
 
 
 class _BaseProxy(Base):
     pass
 
-@jsii.data_type(jsii_type="@scope/jsii-calc-base.BaseProps", jsii_struct_bases=[scope.jsii_calc_base_of_base.VeryBaseProps])
-class BaseProps(scope.jsii_calc_base_of_base.VeryBaseProps, jsii.compat.TypedDict):
-    bar: str
+@jsii.data_type(jsii_type="@scope/jsii-calc-base.BaseProps", jsii_struct_bases=[scope.jsii_calc_base_of_base.VeryBaseProps], name_mapping={'foo': 'foo', 'bar': 'bar'})
+class BaseProps:
+    def __init__(self, *, foo: scope.jsii_calc_base_of_base.Very, bar: str):
```

Comment:
Why do we need the "*"?

Reply:
It makes it so you can't pass any of the following values without using the keyword. Which is required, otherwise adding optional parameters later on might change the order which is breaking.

Reply:
Oh wow. nice!